### DISCUSSION TITLE
How to make one key stroke emit 3 unicode character in a custom keyboard layout?

### DISCUSSION THREAD

**Discussion by ritulahkar** [linuxmint/discussions#349] [Questions]
I have created a custom keyboard layout for my regional language by editing different files. It works fine except for one problem. One of the characters needs 3 unicode characters in combination I.e. I want all the 3 characters to be emitted in single key stroke. Please help me to do so. 

The characters are as follows 
`ক+্+ষ --> ক্ষ`
`U0995 + U09CD + U09B7 --> ক্ষ`
This combination needs to be emitted from the third character of the key `key <AB04>`, shown below.

I added language file to `/usr/share/X11/xkb/symbols` and entry to the `/usr/share/X11/xkb/rules/evdev.xml` file

extract from the file I added:
```
          // shift, z to /
          // key <LFSH> { [ Shift_L ] };
                   key <AB01> { [ U09A2,	U09A7,	U09B7,	voidsymbol ] };
                   key <AB02> { [ U09A3,	U09A8,	U09B8,	voidsymbol  ] };
                   key <AB03> { [ U09AA,	U09AF,	U09B9,	voidsymbol  ] };
                   key <AB04> { [ U09AB,	U09F0,	voidsymbol,	voidsymbol ] };
                   key <AB05> { [ U09AC,	U09B2,	U09DC,	voidsymbol ] };
                   key <AB06> { [ U09AD,	U09F1,	U09DD,	voidsymbol ] };
                   key <AB07> { [ U09AE,	U09B6,	U09DF,	voidsymbol ] };
```
Needs to add the letter ক্ষ to the 3 rd position of 6th key in the shift L line from left.
![image](https://github.com/linuxmint/discussions/assets/105012118/b1b2daf0-6c86-4db5-bd2c-8349c8184e06)

**Comment by mtwebster** [comment]
Hi, this is a very advanced topic, and you probably already know more about doing this than most others.

All I can do is point you to some resources I've recommended in the past:

https://askubuntu.com/questions/510024/what-are-the-steps-needed-to-create-a-new-keyboard-layout
https://www.freedesktop.org/wiki/Software/XKeyboardConfig/
https://www.reddit.com/r/gnome/comments/r4z1p4/how_to_create_custom_keyboard_layout/
